### PR TITLE
Fixed wrong exit code handling by replacing go-shell with the internal sh package

### LINKED ISSUE #68: Wrong exit code upon error in promu release
See https://circleci.com/gh/prometheus/prometheus/3370

```
$ promu release .tarballs
error: could not upload, status code (422 status code 422), msg: Validation Failed, errors: [field: name, code: already_exists]
$ echo $?
0
```

### PR DESCRIPTION
**Pull request by sdurrheimer**
Fixes #68 #8

### CODE REVIEW

File: cmd/crossbuild.go [hunk 1]
```diff
@@ -180,9 +172,17 @@ type platformGroup struct {
 func (pg platformGroup) Build(repoPath string) error {
 	if platformsParam := strings.Join(pg.Platforms[:], " "); platformsParam != "" {
 		fmt.Printf("> running the %s builder docker image\n", pg.Name)
-		if err := docker("run --rm -t -v $PWD:/app", pg.DockerImage, "-i", repoPath, "-p", sh.Quote(platformsParam)); err != nil {
-			return err
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return errors.Wrapf(err, "Couldn't get current working directory")
```
Comment (grobie) [inline cmd/crossbuild.go]: In old-school error handling, messages were supposed to start lowercase. Has this changed with Wrapf?

File: cmd/crossbuild.go [hunk 2]
```diff
@@ -180,9 +172,17 @@ type platformGroup struct {
 func (pg platformGroup) Build(repoPath string) error {
 	if platformsParam := strings.Join(pg.Platforms[:], " "); platformsParam != "" {
 		fmt.Printf("> running the %s builder docker image\n", pg.Name)
```
Comment (grobie) [inline cmd/crossbuild.go]: It's idiomatic in go to reduce indentation levels as much as possible. If I unserstand the method correctly, you could do the following:

```go
platformsParam := strings.Join(pg.Platforms[:], " ")
if len(platformsParam) == 0 {
    return nil
}

fmt.Printf("> running the ...")
// ...
```

File: util/sh/sh.go [hunk 3]
```diff
@@ -47,3 +48,8 @@ func SplitParameters(s string) []string {
 	r := regexp.MustCompile(`'[^']*'|[^ ]+`)
 	return r.FindAllString(s, -1)
 }
+
+// Path joins filepath parts.
+func Path(parts ...string) string {
```
Comment (grobie) [inline util/sh/sh.go]: What's the purpose of this function? Why not just using filepath.Join directly?

**Comment by sdurrheimer** [comment]
@grobie updated

**Pull request merged**